backend/corpora/parliament/utils/parlamint_v4.py

```diff
@@ -0,0 +1,322 @@
+    forename = node.persName.forename.text.strip()
+    name = ' '.join([forename, surname])
+    role = node.persName.roleName.text.strip() if node.persName.roleName else None #too simple, needs to be able to have different values and be gotten from the org_node
+    gender = node.sex['value'].strip() if node.sex else None
+
+    #get org id
+    is_org_node = lambda node: node.name in ['affliation', 'affiliation'] and node.has_attr('ref')
+    org_nodes = node.find_all(is_org_node)
+    org_ids = [org_node['ref'] if org_node else None for org_node in org_nodes]
+    birth_date = node.birth['when'] if node.birth else None
+    birth_year = int(birth_date[:4]) if birth_date else None
+    birthplace = node.birth.placeName.text.strip() if node.birth and node.birth.placeName else None
+
+    wikimedia_uri_node = node.find('idno', type='URI', subtype='wikimedia')
+    wikimedia_uri = wikimedia_uri_node.text if wikimedia_uri_node else None
+
+    twitter_uri_node = node.find('idno', type='URI', subtype='twitter')
+    twitter_uri = twitter_uri_node.text if twitter_uri_node else None
+
+    return {
+        'id': id,
+        'name': name,
+        'role': role,
+        'gender': gender,
+        'org_ids': org_ids,
+        'org_nodes': org_nodes,  # unsure about this still: might be a bit of a memory leak to store the nodes in the metadata
+        'birth_year': birth_year,
+        'birthplace': birthplace,
+        'wikimedia': wikimedia_uri,
+        'twitter': twitter_uri,
+    }
+
+def extract_people_data(soup):
+    person_nodes = soup.find_all('person')
+    person_data = map(extract_person_data, person_nodes)
+    return {
+       person['id']: person for person in person_data
+    }
+
+def current_party_id_extractor():
+    """Extractor that finds the current party, given a date
+    if no date is given, it return the last party in the node"""
+    return Combined(
+        person_attribute_extractor('id'),
+        Metadata('persons'),
+        Metadata('date'),
+        transform=transform_current_party_id
+    )
+
+def organisation_attribute_extractor(attribute):
+    """Extractor that finds the speaker's party and party's
+    attributes defined in extract_party_data()"""
+    return Combined(
+        current_party_id_extractor(),
+        Metadata('organisations'),
+        transform = metadata_attribute_transform_func(attribute),
+    )
+
+def node_is_current(node, date):
+    """Checks if the node is current at the given date
+    i.e. if the date is between the from and to dates of the node"""
+    if node and date:
+        start_date = node.get('from', None)
+        end_date = node.get('to', None)
+        if (start_date and end_date and start_date <= date <= end_date) or \
+        (start_date and start_date <= date) or \
+        (end_date and end_date >= date):
+            return True
+        else:
+            return False
+    else:
+        return False
+
+def transform_current_party_id(data):
```

Comment:
This could do with a docstring. What is `data`? What is is transformed into?

Based on the name, I assumed this function transformed the current party ID into something else, but based on the code, it looks like this function retrieves the current party ID from other data? If so, the name is a bit misleading.